tests/scripts/thread-cert/Cert_9_2_02_MGMTCommissionerSet.py

```diff
@@ -0,0 +1,183 @@
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+from ipaddress import ip_address
+import unittest
+
+from mle import NetworkData
+from network_data import CommissioningData
+import command
+import config
+import mesh_cop
+import mle
+import node
+
+COMMISSIONER = 1
+LEADER = 2
+
+
+class Cert_9_2_02_MGMTCommissionerSet(unittest.TestCase):
+
+    def setUp(self):
+        self.simulator = config.create_default_simulator()
+
+        self.nodes = {}
+        for i in range(1,3):
+            self.nodes[i] = node.Node(i, simulator=self.simulator)
+
+        self.nodes[COMMISSIONER].set_panid(0xface)
+        self.nodes[COMMISSIONER].set_mode('rsdn')
+        self.nodes[COMMISSIONER].add_whitelist(self.nodes[LEADER].get_addr64())
+        self.nodes[COMMISSIONER].enable_whitelist()
+        self.nodes[COMMISSIONER].set_router_selection_jitter(1)
+
+        self.nodes[LEADER].set_panid(0xface)
+        self.nodes[LEADER].set_mode('rsdn')
+        self.nodes[LEADER].add_whitelist(self.nodes[COMMISSIONER].get_addr64())
+        self.nodes[LEADER].enable_whitelist()
+        self.nodes[LEADER].set_router_selection_jitter(1)
+
+    def tearDown(self):
+        for node in list(self.nodes.values()):
+            node.stop()
+            node.destroy()
+        self.simulator.stop()
+
+    def test(self):
+        self.nodes[LEADER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        self.nodes[COMMISSIONER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[COMMISSIONER].get_state(), 'router')
+
+        # Skip all other Coaps sent by Leader
+        self.skip_coap_messages(COMMISSIONER, '0.02')
+        self.skip_coap_messages(LEADER, '2.04')
+
+        # Commissioner start
+        self.nodes[COMMISSIONER].commissioner_start()
+        self.simulator.go(3)
+        self.skip_coap_messages(COMMISSIONER, '0.02')    # Skip LEAD_PET.req
+
+        # Get CommissionerSesssionId from LEAD_PET.rsp
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_coap_message('2.04', assert_enabled=False)
+        commissioner_session_id_tlv = command.get_sub_tlv(msg.coap.payload, mesh_cop.CommissionerSessionId)
+
+        # Step 2 - Harness instructs commissioner to send MGMT_COMMISSIONER_SET.req to Leader
+        steering_data_tlv = mesh_cop.SteeringData(bytes([0xFF]))
+        self.nodes[COMMISSIONER].commissioner_mgmtset_with_tlvs([steering_data_tlv])
+        self.simulator.go(5)
+
+        # Step 3 - Leader responds to MGMT_COMMISSIONER_SET.req with MGMT_COMMISSIONER_SET.rsp
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_coap_message('2.04')
+        command.check_payload_same(msg.coap.payload, (mesh_cop.State(mesh_cop.MeshCopState.REJECT),)) # (mesh_cop.State(mesh_cop.MeshCopState.REJECT),) <- this a tuple, don't delete the comma
```

Comment:
I would prefer simply encapsulating one method, pass the required criteria here. E.g.

```
check_mesh_cop(msg, 
         dest = None,
         condition = [] # required validation
                                 # in (type) format if only check the tlv exist or 
                                 # (type, value) format if need to validate the content additionally
 )
```

Call samples:

```
check_mesh_cop(msg, [(STATE, MeshCopState.REJECT)])
check_mesh_cop(msg, [(STEERING_DATA), (COMMISSIONER_SESSION_ID)])
```